### PR TITLE
Adjust the tone of the Panache/Repository guide for JPA and Mongo

### LINKED ISSUE #326: Panache repository guide comes across as condescending....
Using wording like "Look, we get it: you have a love/hate relationship with DAOs/Repositories but you can’t live without them. We don’t judge, we know life is tough and we’ve got you covered." and "We’re not judging" is off-putting and should probably be avoided. If you're "not judging", then it shouldn't sound like you're judging.

### PR DESCRIPTION
**Pull request by InfoSec812**
Resolves https://github.com/quarkusio/quarkusio.github.io/issues/326

The verbiage in the Repository section of the Panache guide, while explicitly stating "we're not judging", comes across as judgemental/condescending to some people. This is a minor adjustment to the tone of the language used so it can appeal to a broader audience.

### CODE REVIEW

File: docs/src/main/asciidoc/mongodb-panache-guide.adoc [hunk 1]
```diff
@@ -425,8 +428,8 @@ public long count(){
 ----
 
 So if Repositories are your thing, you can keep doing them. Even with repositories, you can keep your entities as
-subclasses of `PanacheMongoEntity` in order to get the ID and public fields working, but you can even skip that and
-go back to specifying your ID and using getters and setters if that's your thing. We're not judging.
+subclasses of `PanacheEntity` in order to get the ID and public fields working, but you can even skip that and
```
Comment (machi1990) [inline docs/src/main/asciidoc/mongodb-panache-guide.adoc]:
```suggestion
subclasses of `PanacheMongoEntity ` in order to get the ID and public fields working, but you can even skip that and
```

Comment (loicmathieu) [inline docs/src/main/asciidoc/mongodb-panache-guide.adoc]: subclasses of `PanacheMongoEntity`

File: docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc [hunk 2]
```diff
@@ -333,8 +333,10 @@ Every query operation accepts passing parameters by index (`Object...`), or by n
 
 == The DAO/Repository option
 
-Look, we get it: you have a love/hate relationship with DAOs/Repositories but you can't live without them. We don't judge, we
-know life is tough and we've got you covered.
+Repositories can be amazing or evil, depending on the complexity of your needs. When they work, 
+they can save you from writing lots of boilerplate and error-prone JPA/SQL/Criteria queries, 
+but when they fail you need a way to drop back to more traditional methods... Don't worry, 
+Panache and Quarkus have you covered either way.
```
Comment (loicmathieu) [inline docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc]: The first sentence feels a bit long and still has some judgment. Maybe going straigth to the point should be bettre.

> Repository is a very popular design and can be very accurate for some use case, Panache and Quarkus have you covered ...

**Review by loicmathieu — CHANGES_REQUESTED** [review]
One error and maybe the sentence about repository can be simplified

File: docs/src/main/asciidoc/mongodb-panache-guide.adoc (hunk 1 above)
Comment (InfoSec812) [reply docs/src/main/asciidoc/mongodb-panache-guide.adoc]: Fixed. Thanks for the review!

File: docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc (hunk 2 above)
Comment (InfoSec812) [reply docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc]: I like it! Updated.

Comment (loicmathieu) [reply docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc]: Still thinking that the following sentence is not needed:
> but when you need to support a complex use case you can revert
to more traditional methods...

And the Don't worry seems to miss some introduction
> If you need them, don't worry ...

**Comment by geoand** [comment]
@InfoSec812 Thanks for the PR! One thing we usually ask contributors to do is squash so a single commit (or into more logical commits if it makes sense) before merging in order to get rid of commits addressing review comments and such.

File: docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc [hunk 3]
```diff
@@ -333,8 +333,10 @@ Every query operation accepts passing parameters by index (`Object...`), or by n
 
 == The DAO/Repository option
 
-Look, we get it: you have a love/hate relationship with DAOs/Repositories but you can't live without them. We don't judge, we
-know life is tough and we've got you covered.
+Repository is a very popular pattern and can be very accurate for some use case, depending on
+the complexity of your needs. When your use case fits, they can save you from writing lots of boilerplate
```
Comment (gsmet) [inline docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc]: What is `they`?

File: docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc [hunk 4]
```diff
@@ -376,7 +378,7 @@ public long count(){
 
 So if Repositories are your thing, you can keep doing them. Even with repositories, you can keep your entities as
 subclasses of `PanacheEntity` in order to get the ID and public fields working, but you can even skip that and
-go back to specifying your ID and using getters and setters if that's your thing. We're not judging.
+go back to specifying your ID and using getters and setters if that's your thing. Use what works for you.
```
Comment (gsmet) [inline docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc]: +1

File: docs/src/main/asciidoc/mongodb-panache-guide.adoc [hunk 5]
```diff
@@ -384,7 +384,10 @@ will use the entity field name: `name`, and native query will use the MongoDB fi
 
 == The DAO/Repository option
 
-Look, we get it: you have a love/hate relationship with DAOs/Repositories but you can't live without them. We don't judge, we know life is tough and we've got you covered.
+Repository is a very popular pattern and can be very accurate for some use case, depending on
+the complexity of your needs. When your use case fits, they can save you from writing lots of boilerplate
```
Comment (gsmet) [inline docs/src/main/asciidoc/mongodb-panache-guide.adoc]: Same question :).

**Review by gsmet — CHANGES_REQUESTED** [review]
Can you have a look at my comments? Thanks!

File: docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc (hunk 3 above)
Comment (gsmet) [reply docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc]: Reading from the comments, I think the first sentence was modified and the second sentence was not and is definitely not accurate anymore.

**Comment by InfoSec812** [comment]
> @InfoSec812 Thanks for the PR! One thing we usually ask contributors to do is squash so a single commit (or into more logical commits if it makes sense) before merging in order to get rid of commits addressing review comments and such.

@geoand I've never really developed a habit of using rebase/squash because I prefer the added details in my commit history. Is there a command which you recommend to make it simple?

**Comment by geoand** [comment]
@InfoSec812 I usually use IntelliJ's build in UI which makes things super easy. Otherwise git's interactive rebase (as explained [here](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#interactive-rebase) for example) is what I occasionally use

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
I made a few additional adjustments and rebased. I think it's good to go now but will wait for CI.

**Pull request merged**